### PR TITLE
#4 added the possibility to use a function as the input to text with …

### PR DESCRIPTION
**Pull request by bjartebore**
…a callback method for hideToast, then we can provide our own logic for hiding toasts

### CODE REVIEW

File: src/Toast.js [hunk 1]
```diff
@@ -40,6 +44,11 @@ class Toast extends Component {
     }
   }
 
+  componentWillUnmount () {
+    const { timeoutId } = this.state;
+    clearTimeout(timeoutId)
+  }
```
Comment (alanshaw) [inline src/Toast.js:50]: @bjartebore what does this do - is it a fix for #11?

File: src/Toast.js [hunk 2]
```diff
@@ -75,17 +84,29 @@ class Toast extends Component {
...
-        </View>
+      textContent = (
+        <TouchableWithoutFeedback onPress={this.onPress}>
+          <View style={styles.container}>
+            <Text style={styles.text}>{text}</Text>
+          </View>
+        </TouchableWithoutFeedback>
+      )
+    } else if (Object.prototype.toString.call(text) === '[object Function]') {
+      textContent = text({ hide: this.hideToast });
```
Comment (alanshaw) [inline src/Toast.js:99]: Please remove semicolon so the linting rules pass

**Review by alanshaw — CHANGES_REQUESTED** [review]
This looks great 🚀 thanks @bjartebore !

Could you please respond to the review queries and also update the docs on the README?

**Comment by alanshaw** [comment]
Also, apologies for not looking at this sooner, for some reason notifications were switched off for me in this repo.

**Comment by ghost** [comment]
@bjartebore still working on this?

**Pull request closed without merging**